src/tasks.cpp

```diff
@@ -30,29 +30,26 @@ void Dispatcher::threadMain()
 		tmpTaskList.swap(taskList);
 		taskLockUnique.unlock();
 
-		for (Task* task : tmpTaskList) {
+		for (auto& task : tmpTaskList) {
 			if (!task->hasExpired()) {
 				++dispatcherCycle;
 				// execute it
 				(*task)();
```

Comment:
While at it, what do you think of writing
```suggestion
				std::invoke(*task);
```
here? I'm not a fan of the `(*ptr)()` syntax because of the extra parenthesis needed 😆